### PR TITLE
fix(insights): Dont throw client_query_id exception in production

### PR DESCRIPTION
**Pull request by timgl**
## Problem

Fix https://sentry.io/organizations/posthog2/issues/3533844597/?project=1899813&referrer=slack

## Changes

- Don't throw exception, just log the warning in sentry

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: posthog/client.py [hunk 1]
```diff
@@ -137,7 +139,11 @@ def validate_client_query_id(
     if not client_query_id:
         return None
     if not client_query_team_id and (not args or "team_id" not in args):
-        raise Exception("Query needs to have a team_id arg if you've passed client_query_id")
+        if DEBUG or TEST:
+            raise Exception("Query needs to have a team_id arg if you've passed client_query_id")
```
Comment (neilkakkar) [inline posthog/client.py:143]: lacking context here, is it possible to pass in a `client_query_id` but not `team_id` in production?

File: posthog/client.py [hunk 2]
```diff
@@ -38,6 +39,7 @@
     CLICKHOUSE_VERIFY,
     TEST,
 )
+from posthog.settings.base_variables import DEBUG
```
Comment (neilkakkar) [inline posthog/client.py:42]:
```suggestion
from django.conf import settings
```

and then `settings.DEBUG`, or `settings.TEST`.

This is the recommended way of accessing django settings & allows easy hooks in tests using django settings overrides.

File: posthog/client.py (hunk 1 above)
Comment (timgl) [reply posthog/client.py]: Grabbing the team_id from the args was the wrong solution. Now you just need to set it explicitly

**Review by neilkakkar — APPROVED** [review]
Much nicer (except the failing tests)

**Pull request merged**